### PR TITLE
docs: add KGPE-D16 connection and setup

### PR DESCRIPTION
**Pull request by miczyg1**
Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

### CODE REVIEW

File: docs/variants/asus_kgpe_d16/setup.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+Power to the Asus KGPE-D16 is delivered by common PC power supply. Simply
+connect the power supply to ATX compatible connectors on the mainboard
+(SSIPWR1, SSI12V1, SSI12V2).
+
+#### SPI
+
+BIOS SPI flash is a socketed DIP-8. There is no simple way to attach to it with
+any tools. That is why we have thought about the
+[qspimux](https://github.com/felixheld/qspimux) made by Felix Held which is
```
Comment (pietrushnic) [inline docs/variants/asus_kgpe_d16/setup.md:16]: @miczyg1 I'm not sure if this section should not be in transparent validation somewhere and just linked here. So extensive description is important, but it does not belong to KGPE-D16 IMO.

Comment (miczyg1) [reply docs/variants/asus_kgpe_d16/setup.md:16]: Nowhere else do we use the qspimux besides the KGPE-D16, I don't see any reason to move it elsewhere. qspimux helps to update the flash content and does not have any validation purpose. Moreover qspimux is customizable and can be used with a variety of chips and adapters, so putting it into some generic section will only cause a scatter of information and jumping throughout the repository.

Comment (pietrushnic) [reply docs/variants/asus_kgpe_d16/setup.md:16]: @miczyg1 what about potential reuse of information?

Comment (pietrushnic) [reply docs/variants/asus_kgpe_d16/setup.md:16]: @miczyg1 let's agree if any other project will use this tool, then you will move the documentation to generic section, so both hardware platforms will not cross refere itself, but will point to generic content, ok?

**Comment by pietrushnic** [comment]
@macpijan if in long run validation should take over KGPE-D16 hardware maintanance in lab they probably should spent 5min reviewing this document.

File: docs/variants/asus_kgpe_d16/setup.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+ RTE header J10 pin | qspimux J101 pin
+:------------------:|:-------------------:
+ 1 (GPIO400)        | 9 (MUX_SEL)
+ 2 (GPIO401)        | 4 (IO3_HOLD_PROG)
+
+![](/images/qspimux_pin_header.png)
+
+![](/images/qspimux_kgpe.jpeg)
+
+> qspimux schematics: https://github.com/felixheld/qspimux/blob/master/qspimux/qspimux.pdf
```
Comment (pietrushnic) [inline docs/variants/asus_kgpe_d16/setup.md:64]: @miczyg1 please hide links behind markdown [link]()

File: docs/variants/asus_kgpe_d16/setup.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+| RTE header J11 pin     | Asus KGPE-D16             |
+|:----------------------:|:-------------------------:|
+| 8 (OC buffer output)   | PANEL1 pin 11 (PWR)       |
+| 9 (OC buffer output)   | PANEL1 pin 17 (RESET)     |
+
+Additionally connect one of the PANEL1 ground pins to one of the ground pins on
+RTE.
+
+### MANUAL
```
Comment (pietrushnic) [inline docs/variants/asus_kgpe_d16/setup.md:112]:
```suggestion
### Manual
```

File: docs/variants/asus_kgpe_d16/setup.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+For more details about KGPE-D16 refer to the manual:
+https://dlcdnets.asus.com/pub/ASUS/mb/SocketG34(1944)/KGPE-D16/Menual_QVL/E8847_KGPE-D16.pdf
```
Comment (pietrushnic) [inline docs/variants/asus_kgpe_d16/setup.md:115]: @miczyg1 maybe we should drop that to this repo? I'm not sure how long this manual will be available on ASUS website

File: docs/variants/asus_kgpe_d16/setup.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+available with qspimux. Flashing may be done via the commands shown earlier or
+the flash script on present in the root directory of the RTE:
+
+```shell
+./flash.sh coreboot.rom
+```
+
+The script will automatically power off the board and update flash chip
+contents using flashrom. After the operation is finished the routing of SPI
+signals is restored to the platform.
```
Comment (pietrushnic) [inline docs/variants/asus_kgpe_d16/setup.md:137]: @miczyg1 how we will handle OTP and WP? In long run if we plan to implement OTP Winbond will be replaced with something else, shalle we update this document then?

File: docs/variants/asus_kgpe_d16/setup.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+#### Power management
+
+RTE is able to control the power of the board with the `rte_ctrl` application
+or with the web GUI available at RTE's IP address port 8000. Possible command are:
```
Comment (pietrushnic) [inline docs/variants/asus_kgpe_d16/setup.md:142]: @miczyg1 it is worh to mentioned more references to RTE, there are some information on Github as well in documentation. Source code to `rte_ctrl` tool also can be pointed e..g if someone would like to fix/extend the tool.

File: docs/variants/asus_kgpe_d16/setup.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+    echo -e '\n'
+else
+    echo -e "\$1 == on|off|show|toggle\nEdit this script to set the sonoff ip."
+    echo -e 'Current state:'
+    wget -q -O - http://$SONOFF_IP/switch/sonoff_s20_relay
+    echo -e '\n'
+fi
+```
+
+To turn off the pwoer from the platform `./sonoff.sh off` and to turn it on
```
Comment (pietrushnic) [inline docs/variants/asus_kgpe_d16/setup.md:181]:
```suggestion
To turn off the power from the platform `./sonoff.sh off` and to turn it on
```
Please run spell checker

File: docs/variants/asus_kgpe_d16/setup.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+A short demo presenting RTE capabilities with KGPE-D16:
+
+<script id="asciicast-438342" src="https://asciinema.org/a/438342.js" async data-rows="25"></script>
+
+### Remote access
+
+if you are interested in developing coreboot on KGPE-D16 and have some spare
+time, there is a possibility to obtain the access to the hardware. Drop us an
+email to [leads@3mdeb.com](mailto:leads@3mdeb.com) with a request.
```
Comment (pietrushnic) [inline docs/variants/asus_kgpe_d16/setup.md:208]: @miczyg1 in long run we looking to extend this section: https://docs.dasharo.com/inquiries/overview/

File: docs/variants/asus_kgpe_d16/setup.md (hunk 1 above)
Comment (miczyg1) [reply docs/variants/asus_kgpe_d16/setup.md:16]: Potential reuse of information existing in this document is just a few sentences (2-3). If it grows and appears in other systems then I'm fine with that approach.

File: docs/variants/asus_kgpe_d16/setup.md (hunk 5 above)
Comment (miczyg1) [reply docs/variants/asus_kgpe_d16/setup.md:137]: Currently we do not use OTP chip in the setup. As for WP pin, we may control it with a RTE GPIO, just one more wire to connect

File: docs/variants/asus_kgpe_d16/setup.md (hunk 6 above)
Comment (miczyg1) [reply docs/variants/asus_kgpe_d16/setup.md:142]: I believe we do not have the source for rte_ctrl... We may later replace it with a bash script with equivalent functionality

Comment (pietrushnic) [reply docs/variants/asus_kgpe_d16/setup.md:142]: really? https://github.com/3mdeb/RteCtrl

Comment (pietrushnic) [reply docs/variants/asus_kgpe_d16/setup.md:142]: This is the bash for testing: https://github.com/3mdeb/RteCtrl/blob/master/test/rteCtrl.sh

Comment (miczyg1) [reply docs/variants/asus_kgpe_d16/setup.md:142]: Really. This is the RteCtrl service code wit the GUI. `rte_ctrl` is the old command line tool which I could not find source for.

Comment (macpijan) [reply docs/variants/asus_kgpe_d16/setup.md]: Added more RTE references. In terms of rte_ctrl. I guess we can leave it for now, but maybe we should swich to using RteCtrl exclusively.

File: docs/variants/asus_kgpe_d16/setup.md (hunk 7 above)
Comment (macpijan) [reply docs/variants/asus_kgpe_d16/setup.md]: done

File: docs/variants/asus_kgpe_d16/setup.md (hunk 4 above)
Comment (macpijan) [reply docs/variants/asus_kgpe_d16/setup.md]: done

File: docs/variants/asus_kgpe_d16/setup.md (hunk 6 above)
Comment (pietrushnic) [reply docs/variants/asus_kgpe_d16/setup.md]: We should switch to something what code we have.

**Pull request merged**